utils/conversion_functions.py

```diff
@@ -86,7 +84,37 @@ def convert(rules, input_text) -> str:
                 word = unicodedata.normalize("NFC", word)
                 result += word + whitespace_sequence
             output_text = result
-    return output_text
+    return unicodedata.normalize("NFC", output_text)
+
+
+def normalize_replace_rules(rules):
+    """ From a given Replace rule, derives rules with the following modifications:
+        1. A version of the rule with all lower case letters.
+        2. A version of the rule with the initial letter only uppercase.
+        3. All vowels match their accented and unaccented versions,
+            e.g. "a" matches "a" and "á".
+        4. Prevent patterns from matching only part of a sequence of
+           combining diacritics (in other words, h shouldn't match ȟ even
+           though the former is a prefix of the latter because it would
+           leave an orphaned combining diacritic character)
+    """
+    normalized_rules = []
+    for rule_type, target, repl in rules:
+        assert rule_type == "Replace"
+        target = target.lower()
+        repl = repl.lower()
+        target = unicodedata.normalize("NFD", target)
+        repl = unicodedata.normalize("NFD", repl)
+        target = re.sub(r"([aeiou])", r"\1(́)?", target) # a -> a(́)?
+        target += r"(?=[a-zA-Z]|\s|$)"
+        repl = re.sub(r"([aeiou])", r"\1\\1", repl) # a -> a\1
+        normalized_rules.append((rule_type, target, repl))
+        if target:
+            target = target[0].upper() + target[1:]
+        if repl:
+            repl = repl[0].upper() + repl[1:]
+        normalized_rules.append((rule_type, target, repl))

```

Comment:
Needs to be amended to preserve all different kinds of capitalization, not just initial letter capitalization.